source/plugin/modularization.rst

```diff
@@ -0,0 +1,61 @@
+===================
+Code Modularization
+===================
+
+When writing Sponge plugins, you may find that you need to modularize your code. Maybe you want to split your core into
+another project so that others can make use of it, or maybe you want to utilize a service another plugin provides. With
+Sponge, gone are the days of shading entire plugins into your plugin.
+
+Problem 1: Providing Good Service
+=================================
+
+Let's say you want to provide some awesome service for ``WarpService``. You want to make ``AwesomeWarpService`` override
+some other plugin's ``CoreWarpService`` with your own.
+
+But when you go to use ``Sponge.getServiceManager().provide(WarpService.class)``, you are shocked to find that referencing
+``WarpService`` throws a ``ClassNotFoundException``!
+
+What's Happening
+~~~~~~~~~~~~~~~~
+
+When you compile your plugin, the compiler finds ``WarpService.class`` because it's present in the workspace (either via
+a build tool like Gradle or by just adding it to the classpath). However, when Sponge goes to run your plugin, it can't
+find the referenced ``WarpService.class`` in your plugin's "container".
+
+Solution
+~~~~~~~~
+
+Using Sponge's dependency system, you can declare hard and soft dependencies using the ``dependencies`` field of the
```

Comment:
I like that you are covering the dependency system, I'm not sure this has been covered elsewhere yet.